contracts/index_executor/src/types.rs

```diff
@@ -0,0 +1,151 @@
+use alloc::vec::Vec;
+use scale::{Decode, Encode};
+
+/// Definition of source edge
+#[derive(Clone, Decode, Encode, Eq, PartialEq, Ord, PartialOrd, Debug)]
+#[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+pub struct SourceEdge {
+    /// asset/chain
+    pub to: Vec<u8>,
+    /// Capacity of the edge
+    pub cap: u128,
+    /// Flow of the edge
+    pub flow: u128,
+    /// Price impact after executing the edge
+    pub impact: u128,
+}
+
+/// Definition of SINK edge
+#[derive(Clone, Decode, Encode, Eq, PartialEq, Ord, PartialOrd, Debug)]
+#[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+pub struct SinkEdge {
+    /// asset/chain
+    pub from: Vec<u8>,
+    /// Capacity of the edge
+    pub cap: u128,
+    /// Flow of the edge
+    pub flow: u128,
+    /// Price impact after executing the edge
+    pub impact: u128,
+}
+
+/// Definition of swap operation edge
+#[derive(Clone, Decode, Encode, Eq, PartialEq, Ord, PartialOrd, Debug)]
+#[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+pub struct SwapEdge {
+    /// asset/chain
+    pub from: Vec<u8>,
+    /// asset/chain
+    pub to: Vec<u8>,
+    /// Chain name
+    pub chain: Vec<u8>,
+    /// Dex name
+    pub dex: Vec<u8>,
+    /// Capacity of the edge
+    pub cap: u128,
+    /// Flow of the edge
+    pub flow: u128,
+    /// Price impact after executing the edge
+    pub impact: u128,
+    /// Original relayer account balance of spend asset
+    pub b0: Option<u128>,
+    /// Original relayer account balance of received asset
+    pub b1: Option<u128>,
+}
+
+/// Definition of bridge operation edge
+#[derive(Clone, Decode, Encode, Eq, PartialEq, Ord, PartialOrd, Debug)]
+#[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+pub struct BridgeEdge {
+    /// asset/chain
+    from: Vec<u8>,
+    /// asset/chain
+    to: Vec<u8>,
+    /// Capacity of the edge
+    cap: u128,
+    /// Flow of the edge
+    flow: u128,
+    /// Price impact after executing the edge
+    impact: u128,
+    /// Original relayer account balance of asset on source chain
+    b0: Option<u128>,
+    /// Original relayer account balance of asset on dest chain
+    b1: Option<u128>,
+}
+
+#[derive(Clone, Decode, Encode, Eq, PartialEq, Ord, PartialOrd, Debug)]
+#[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+pub enum EdgeStatus {
```

Comment:
Better to not use "edge" refer to a step. Use "step", "action" or "transaction" instead.